### PR TITLE
Added possibility to add `auth` in model Meta

### PR DESCRIPTION
**Pull request by phantom-as**
### Status
**READY**

### Background context

After adding auth to the client, pushing to the model.

### Description of changes

Expose auth on models' Meta class, an translated it to the client through the manager.

### Checklist

- [x] I provided a detailed description of the change, including motivations and implementation details
- [x] I have performed a self-review of my own code
- [x] I implemented unit tests for all new or modified behaviours
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, CHANGELOG and VERSION

### CODE REVIEW

File: django_json_api/manager.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def get_base_lookup_to_nested_lookups_mapping(lookups: List[str]) -> Dict[str, L
 
 class JSONAPIManager:
     def __init__(self, model, **kwargs):
-        self.client = JSONAPIClient()
+        self.client = JSONAPIClient(auth=model._meta.auth)
```
Comment (aubryreveal) [inline django_json_api/manager.py:24]: you access the private _meta attribute.

It is not a great practice but I don't see a clear way around it (except by adding a getter at the model level but I not convinced) .

**Review by aubryreveal — CHANGES_REQUESTED** [review]

File: tests/test_manager.py [hunk 2]
```diff
@@ -74,6 +74,11 @@ def empty_page() -> Mocker:
         yield mocker
 
 
+def test_jsonapi_manager_uses_auth_from_model() -> None:
+    manager = JSONAPIManager(DummyWithAuth)
+    assert manager.client.session.auth == DummyWithAuth._meta.auth
```
[the remark this answers is not in the document]
Comment (phantom-as) [reply tests/test_manager.py:79]: I disagree. What it tests is that the client is indeed instanciated with the right "auth" param.

File: django_json_api/manager.py (hunk 1 above)
Comment (phantom-as) [reply django_json_api/manager.py:24]: While I agree on principle, this very "django" style. If you look in Django's code, it is happening very often.
What we did from day 1 with this lib is stick close to what Django is doing, to it makes sense IMHO.

Comment (aubryreveal) [reply django_json_api/manager.py:24]: Yeah I figure ^^

File: tests/test_manager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aubryreveal) [reply tests/test_manager.py:79]: Yes, it is what I find "not much".

But it is arguable, so you can leave it there.

**Review by aubryreveal — APPROVED** [review]

**Pull request merged**